hazelcast/src/main/java/com/hazelcast/internal/util/ThreadLocalRandomProvider.java

```diff
@@ -38,7 +40,8 @@ private ThreadLocalRandomProvider() {
     public static Random get() {
         Random random = THREAD_LOCAL_RANDOM.get();
         if (random == null) {
-            random = new Random();
+            long seed = SEED_GENERATOR.nextLong();
```

Comment:
in Sun Java 6 the seed generator is not thread-safe: 2 threads can create 2 instances of `Random` initialized with the same seed. 

However when you have a single `Random` instance then 2 threads calling `getLong()` will get different results. The CAS loop is inside `Random`. It has been like this since at least Java 5. See http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6362070